internal/provider/resource_definition_resource_test.go

```diff
@@ -119,6 +120,48 @@ func TestAccResourceDefinition(t *testing.T) {
 			},
 			resourceAttrNameUpdateValue2: "test-2",
 		},
+		{
+			name: "S3 static - secret refs",
+			configCreate: func() string {
+				return testAccResourceDefinitionS3taticResourceWithSecretRefs("accessKeyIdPath1", "secretAccessKeyPath1")
+			},
+			resourceAttrNameIDValue:      "s3-test-with-secrets",
+			resourceAttrNameUpdateKey:    "driver_inputs.secret_refs",
+			resourceAttrNameUpdateValue1: "{\"aws_access_key_id\":{\"ref\":\"accessKeyIdPath1\",\"store\":\"external-secret-store\",\"version\":\"1\"},\"aws_secret_access_key\":{\"ref\":\"secretAccessKeyPath1\",\"store\":\"external-secret-store\",\"version\":\"1\"}}",
+			resourceAttrName:             "humanitec_resource_definition.s3_test_with_secrets",
+			configUpdate: func() string {
+				return testAccResourceDefinitionS3taticResourceWithSecretRefs("accessKeyIdPath2", "secretAccessKeyPath2")
+			},
+			resourceAttrNameUpdateValue2: "{\"aws_access_key_id\":{\"ref\":\"accessKeyIdPath2\",\"store\":\"external-secret-store\",\"version\":\"1\"},\"aws_secret_access_key\":{\"ref\":\"secretAccessKeyPath2\",\"store\":\"external-secret-store\",\"version\":\"1\"}}",
+		},
+		{
+			name: "S3 static - secret ref set values",
+			configCreate: func() string {
+				return testAccResourceDefinitionS3taticResourceWithSecretRefValues("accessKeyId1", "secretAccessKey1")
+			},
+			resourceAttrNameIDValue:      "s3-test-with-secrets",
+			resourceAttrNameUpdateKey:    "driver_inputs.secret_refs",
+			resourceAttrNameUpdateValue1: "{\"aws_access_key_id\":{\"value\":\"accessKeyId1\"},\"aws_secret_access_key\":{\"value\":\"secretAccessKey1\"}}",
```

Comment:
We are now storing state as we receive it as input, not what we receive from resources service (unless input is nil).